### PR TITLE
Unwanted conditions to notes schema

### PR DESCRIPTION
**Pull request by lucero-v**
This PR adds a schema for converting unwanted conditions to notes to be used with MARS OTC data

Test Steps:
1. Add the following rules to a receiver transform
```
  - name: observation-result-with-aoe
    resource:
      - "%resource.result.resolve().where(code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code in ('840539006))) | %resource.basedOn.resolve().supportingInfo.resolve()"

  - name: order-note
    resource: "%resource.basedOn.resolve().note | %resource.result.resolve().where(code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code in ('840539006')).exists().not())"
    schema: classpath:/metadata/hl7_mapping/receivers/Common/observation-to-nte/unwanted-conditions-to-notes.yml
```
Here we are indicating that all results that do not have a code of '840539006' (COVID) should go through the new `unwanted-conditions-to-notes` schema

2. Run a multiplex test and verify that all other conditions except for COVID are shown as NTE segments on the resulting message.

## Changes
- added `unwanted-conditions-to-notes` schema
- added integration test

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

### CODE REVIEW

File: prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/mars_unwanted_condition_to_notes.hl7 [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ORC|RE|bfe0d734-06a8-472e-bf6a-cad808150c4e^Testing Lab^12D4567890^CLIA|bfe0d734-06a8-472e-bf6a-cad808150c4e^Testing Lab^12D4567890^CLIA|||||||||1245319599^McTester^Phil^^^^^^NPI&2.16.840.1.113883.4.6&ISO^L^^^NPI||(530) 867 5309^WPN^PH^^1^530^8675309^^^^^(530) 867 5309|20240828174209+0000||||||Testing Lab^L^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^12D4567890|123 Beach Way^^Denver^CO^80210^USA|(530) 867 5309^WPN^PH^^1^530^8675309^^^^^(530) 867 5309|321 Ocean Drive^^Denver^CO^80210^USA
+OBR|1|bfe0d734-06a8-472e-bf6a-cad808150c4e^Testing Lab^12D4567890^CLIA|bfe0d734-06a8-472e-bf6a-cad808150c4e^Testing Lab^12D4567890^CLIA|96123-5^SARS-CoV-2 (COVID-19) RdRp gene [Presence] in Upper respiratory specimen by NAA with probe detection^LN|||20240828172709+0000|||||||||1245319599^McTester^Phil^^^^^^NPI&2.16.840.1.113883.4.6&ISO^L^^^NPI|(530) 867 5309^WPN^PH^^1^530^8675309^^^^^(530) 867 5309|||||20240828174209+0000|||F
+NTE|1||This is a note
+NTE|2|0|OBX filtered for identifier = 96123-5 - COVID-19; value = 260373001 - Detected. OBX was removed due to your jurisdictional reporting rules indicating this result is not reportable.
```
Comment (GilmoreA6) [inline prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/mars_unwanted_condition_to_notes.hl7:7]: This value should not be numeric, This should be one of the values in table HL70105 "O", "L" or "P"

Comment (GilmoreA6) [reply prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/mars_unwanted_condition_to_notes.hl7:7]: Can we also add a test where multiple observations are converted to notes?

**Review by oslynn — APPROVED** [review]

File: prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/mars_unwanted_condition_to_notes.hl7 (hunk 1 above)
Comment (lucero-v) [reply prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/mars_unwanted_condition_to_notes.hl7]: Ah, nice catch, thanks. Also added test with Flu A, Flu B, and COVID.

**Review by GilmoreA6 — APPROVED** [review]
Good Job!

**Pull request merged**